client/src/pages/AdvancedSearch/Form.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { useState } from "react";
+
+export const AdvancedSearchForm = (callback: any, initialState = {}) => {
+  const [values, setValues] = useState(initialState);
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValues({ ...values, [event.target.name]: event.target.value });
+  };
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    await callback();
+  };
+  return {
+    onChange,
+    onSubmit,
+    values,
+  };
+};
```

Comment:
Can't wait to try this out, this could make things easier for other parts of this app if implemented in a similar way, great job!

Reply:
Thank you